cognite/neat/graph/extractors/_iodd.py

```diff
@@ -104,29 +151,59 @@ def _text_elements2triples(cls, et_root: Element, namespace: Namespace) -> list[
                     text_id = namespace[id]
 
                     # Create Text node
-                    triples.append((text_id, RDF.type, as_neat_compliant_uri(IODD["Text"])))
+                    triples.append((text_id, RDF.type, as_neat_compliant_uri(IODD["TextObject"])))
 
                     # Resolve text value related to the text item
                     if value := element.attrib.get("value"):
                         triples.append((text_id, IODD.value, Literal(value)))
 
         return triples
 
-    # TODO
     @classmethod
-    def _process_data_collection2triples(cls, pc_root: Element, namespace: Namespace) -> list[Triple]:
+    def _std_variables2triples(cls, vc_root: Element, namespace: Namespace, device_id: URIRef) -> list[Triple]:
         """
-        ProcessDataCollection contains both ProcessDataIn and ProcessDataOut. Here, we are interested in the elements
-        that corresponds to actual sensor values that we will rececive from the MQTT connection.
+        For simplicity, only extract the two items we want for this use case - V_ApplicationSpecificTag and
+        V_SerialNumber
+        """
+        triples: list[Triple] = []
+
+        if std_variable_elements := get_children(vc_root, child_tag="StdVariableRef", ignore_namespace=True):
+            for element in std_variable_elements:
+                if id := element.attrib.get("id"):
+                    if id in cls.std_variable_elements_to_extract:
+                        predicate = to_camel(id.replace("V_", ""))
+                        object = element.attrib.get("defaultValue", "")
+                        triples.append((device_id, IODD[predicate], Literal(object)))
+        return triples
 
-        Most likely this is ProcessDataOut elements (to be verified).
+    @classmethod
+    def _variables_data_collection2triples(
+        cls, vc_root: Element, namespace: Namespace, device_id: URIRef
+    ) -> list[Triple]:
+        """
+        VariableCollection contains elements that references Variables and StdVariables. The StdVariables
+        can be resolved by looking up the ID in the IODD-StandardDefinitions1.1.xml sheet.
+
+        The Variable elements are descriptions of the sensors collecting data for the device.
         """
         triples: list[Triple] = []
 
+        # StdVariableRef elements of interest
+        triples.extend(cls._std_variables2triples(vc_root, namespace, device_id))
+
+        # Variable elements (these are the descriptions of the sensors)
+        if variable_elements := get_children(vc_root, child_tag="Variable", ignore_namespace=True):
+            for element in variable_elements:
+                if id := element.attrib.get("id"):
+                    variable_id = f"{device_id}_{id}"
```

Comment:
I guess this will be heavy influenced how ids in future MQTT will look like, or how they can be created.
I expect that we will in future extend configuration of Iodd extractor that configures how variable_id are made

Reply:
yes.. I think I will just try with this first and then we can reiterate. As of now they will not even use MQTT so we have some flexibility in how we define the time series